src/main/java/com/snowflake/kafka/connector/internal/streaming/BufferedTopicPartitionChannel.java

```diff
@@ -915,6 +925,7 @@ private void resetChannelMetadataAfterRecovery(
       // Set the flag so that any leftover rows in the buffer should be skipped, it will be
       // re-ingested since the offset in kafka was reset
       needToSkipCurrentBatch = true;
+      this.channel = newChannel;
```

Comment:
Is this correct? At line 904 above, we simply return if the channel offset is null, this means that we will never use the new channel if the offset is null?